dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueChannel.java

```diff
@@ -242,6 +242,8 @@ private static ImmutableList<LimitedChannel> createHostChannels(
                 channel = HostMetricsChannel.create(cf, channel, targetUri.uri());
                 channel =
                         new TraceEnrichingChannel(channel, DialogueTracing.tracingTags(cf, uriIndexForInstrumentation));
+                channel = new DeadlineAdvertisementChannel(
```

Comment:
are there other places we need to create one of these?

Reply:
There shouldn't be -- this should do the trick! This position looks great because executes after the queues and retrying channels so that it will be re-applied on each request with an updated `Expect-With` value